src/openeo_gfmap/manager/job_manager.py

```diff
@@ -150,24 +202,40 @@ def _resume_postjob_actions(self, df: pd.DataFrame):
             job = connection.job(row.id)
             if row.status == "postprocessing":
                 _log.info(
-                    f"Resuming postprocessing of job {row.id}, queueing on_job_finished..."
+                    "Resuming postprocessing of job %s, queueing on_job_finished...",
+                    row.id,
+                )
+                future = self._executor.submit(self.on_job_done, job, row, _stac_lock)
+                future.add_done_callback(
+                    partial(
+                        done_callback,
+                        df=df,
+                        idx=idx,
+                    )
                 )
-                future = self._executor.submit(self.on_job_done, job, row)
-                future.add_done_callback(partial(done_callback, df=df, idx=idx))
             else:
                 _log.info(
-                    f"Resuming postprocessing of job {row.id}, queueing on_job_error..."
+                    "Resuming postprocessing of job %s, queueing on_job_error...",
+                    row.id,
                 )
                 future = self._executor.submit(self.on_job_error, job, row)
-                future.add_done_callback(partial(done_callback, df=df, idx=idx))
+                future.add_done_callback(
+                    partial(
+                        done_callback,
+                        df=df,
+                        idx=idx,
+                    )
+                )
             self._futures.append(future)
 
     def _restart_failed_jobs(self, df: pd.DataFrame):
         """Sets-up failed jobs as "not_started" as they will be restarted by the manager."""
         failed_tasks = df[df.status.isin(["error", "start_failed"])]
         not_started_tasks = df[df.status == "not_started"]
         _log.info(
-            f"Resetting {len(failed_tasks)} failed jobs to 'not_started'. {len(not_started_tasks)} jobs are already 'not_started'."
+            "Resetting %s failed jobs to 'not_started'. %s jobs are already 'not_started'.",
+            len(failed_tasks),
+            len(not_started_tasks),
         )
         for idx, _ in failed_tasks.iterrows():
             df.loc[idx, "status"] = "not_started"
```

Comment:
in the latest version of the job manager update statuses will be changed to track statuses

Reply:
Not sure how to handle that. Is the `_update_statuses` function going to be removed from one version to the other?

Reply:
I propose we open an issue to solve this whenever the new OpenEO Python client release